### PR TITLE
n2p2: Add new package

### PR DESCRIPTION
**Pull request by kuramoto-fj**
Add new package : n2p2
This repository provides ready-to-use software for high-dimensional neural network potentials in computational physics and chemistry.

### CODE REVIEW

**Comment by alalazo** [comment]
Were the patches submitted upstream? Does it make sense to add also v2.1.1?

**Comment by kuramoto-fj** [comment]
> Were the patches submitted upstream? Does it make sense to add also v2.1.1?

Thank you for your reply.
I upstreamed the patches. And I added v2.1.1 on the receipe.

**Review by alalazo — DISMISSED** [review]

File: var/spack/repos/builtin/packages/n2p2/package.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    patch("libnnp-makefile.patch")
+    patch("nnp_test.h.patch")
+
+    depends_on("mpi")
+    depends_on("blas")
+    depends_on("gsl", type=("build", "run", "link"))
+    depends_on("eigen", type=("build", "run", "link"))
+    depends_on("doxygen", type="build", when="+doc")
+    depends_on("texlive", type="build", when="+doc")
+    depends_on("py-cython", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/n2p2/package.py:39]: This is usually build-only

File: var/spack/repos/builtin/packages/n2p2/package.py [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    depends_on("py-breathe", type=("build", "run"), when="+doc")
+    depends_on("py-sphinx", type=("build", "run"), when="+doc")
+    depends_on("py-sphinx-rtd-theme", type=("build", "run"), when="+doc")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/n2p2/package.py]: Doc deps should be build-only

File: var/spack/repos/builtin/packages/n2p2/package.py [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+    depends_on("py-breathe", type=("build", "run"), when="+doc")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/n2p2/package.py]: This line is duplicated

File: var/spack/repos/builtin/packages/n2p2/package.py [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+
+    depends_on("boost", type=("build", "run", "link"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/n2p2/package.py]: `boost` doesn't have a `prefix.bin` directory, so the build/run types won't do anything. This should probably be `type='link'` or just use the default.

File: var/spack/repos/builtin/packages/n2p2/package.py [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+    depends_on("lcov", type=("build", "run"))
+    depends_on("py-pytest", type=("build", "run"))
+    depends_on("py-pytest-cov", type=("build", "run"))
+    depends_on("py-coverage@4.5.4", type=("build", "run"))
+    depends_on("py-packaging", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/n2p2/package.py]: Are these actually needed at build/run time or just to run the tests?

File: var/spack/repos/builtin/packages/n2p2/package.py (hunk 1 above)
Comment (kuramoto-fj) [reply var/spack/repos/builtin/packages/n2p2/package.py:39]: I tried it. But I got a error of cypthon in spack test.
The error was "ModuleNotFoundError: No module named 'Cython'".
So I could not change type=("build", "run") to type="build"

File: var/spack/repos/builtin/packages/n2p2/package.py (hunk 2 above)
Comment (kuramoto-fj) [reply var/spack/repos/builtin/packages/n2p2/package.py]: Sorry. I missed.

File: var/spack/repos/builtin/packages/n2p2/package.py (hunk 3 above)
Comment (kuramoto-fj) [reply var/spack/repos/builtin/packages/n2p2/package.py]: Sorry. I removed the line.

File: var/spack/repos/builtin/packages/n2p2/package.py (hunk 4 above)
Comment (kuramoto-fj) [reply var/spack/repos/builtin/packages/n2p2/package.py]: I agree.
I changed type=("build", "run", "link") to type="link".

File: var/spack/repos/builtin/packages/n2p2/package.py (hunk 5 above)
Comment (kuramoto-fj) [reply var/spack/repos/builtin/packages/n2p2/package.py]: The n2p2 receipe use these oss just to run the test.
I've tried those combinations and found a pattern that works, so I'll change it.
Py-test's type is ("build", "run"), but other's type are "build".

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**